[EP02]날씨 좋은 월요일 오후 세 시, 자전거 타는 사람은 몇 명?


##목차##

#1_프로젝트 1 

    (1) 데이터 가져오기
    (2) 모델에 입력할 데이터 X 준비하기
    (3) 모델에 예측할 데이터 y 준비하기
    (4) train 데이터와 test 데이터로 분리하기
    (5) 모델 준비하기
입력 데이터 개수에 맞는 가중치 W와 b를 준비해주세요.
모델 함수를 구현해주세요.

(6) 손실함수 loss 정의하기
손실함수를 MSE 함수로 정의해주세요.

(7) 기울기를 구하는 gradient 함수 구현하기
기울기를 계산하는 gradient 함수를 구현해주세요.

(8) 하이퍼 파라미터인 학습률 설정하기
학습률, learning rate 를 설정해주세요
만약 학습이 잘 되지 않는다면 learning rate 값을 한번 여러 가지로 설정하며 실험해 보세요.

(9) 모델 학습하기
정의된 손실함수와 기울기 함수로 모델을 학습해주세요.
loss값이 충분히 떨어질 때까지 학습을 진행해주세요.
입력하는 데이터인 X에 들어가는 특성 컬럼들을 몇 개 빼도 괜찮습니다. 다양한 데이터로 실험해 보세요.

(10) test 데이터에 대한 성능 확인하기
test 데이터에 대한 성능을 확인해주세요.

(11) 정답 데이터와 예측한 데이터 시각화하기
x축에는 X 데이터의 첫 번째 컬럼을, y축에는 정답인 target 데이터를 넣어서 모델이 예측한 데이터를 시각화해 주세요.


#2_프로젝트 2 :

(1) 데이터 가져오기
터미널에서 ~/data/data/bike-sharing-demand 경로에 train.csv 데이터를 train 변수로 가져 옵니다.
여기에서 데이터를 상세히 확인할 수 있습니다.
(2) datetime 컬럼을 datetime 자료형으로 변환하고 연, 월, 일, 시, 분, 초까지 6가지 컬럼 생성하기
to_datetime 활용 : 링크 참조
pandas datetime extract year month day 키워드로 검색해 보기
(3) year, month, day, hour, minute, second 데이터 개수 시각화하기
sns.countplot 활용해서 시각화하기
subplot을 활용해서 한 번에 6개의 그래프 함께 시각화하기
(4) X, y 컬럼 선택 및 train/test 데이터 분리
X(feature) 에는 어떤 데이터 컬럼을 넣어야 될지 고민해 봅시다 데이터 컬럼을 추가하거나 빼면서 잘 조합하여 데이터를 넣기 바랍니다. 만일 모든 데이터 컬럼(특징)을 넣는다면 오차 값이 말도 안 되게 적게 나올 수도 있습니다. ex) count를 예측하는데 casual,registered 컬럼이 필요할까요?

count 값을 맞추고자 하므로, y 변수에 count 컬럼의 데이터 넣기

(5) LinearRegression 모델 학습
sklearn의 LinearRegression 모델 불러오기 및 학습하기
(6) 학습된 모델로 X_test에 대한 예측값 출력 및 손실함수값 계산
학습된 모델에 X_test를 입력해서 예측값 출력하기
모델이 예측한 값과 정답 target 간의 손실함수 값 계산하기
mse 값과 함께 rmse 값도 계산하기
(7) x축은 temp 또는 humidity로, y축은 count로 예측 결과 시각화하기
x축에 X 데이터 중 temp 데이터를, y축에는 count 데이터를 넣어서 시각화하기
x축에 X 데이터 중 humidity 데이터를, y축에는 count 데이터를 넣어서 시각화하기



3_회고 :

    1)이번 프로젝트에서 어려웠던 점
    2)프로젝트를 진행하면서 본인이 학습한 내용
    3)프로젝트를 진행하면서 알아낸 점 혹은 아직 모호한 점
    4)루브릭 평가 지표를 맞추기 위해 시도한 것들
    5)만약에 루브릭 평가 지표를 달성 하지 못했을 때, 이유에 관한 추정
    6)자기 다짐




In [35]:
#(1)데이터 가져오기
#sklearn.datasets의 load_diabetes에서 데이터를 가져와주세요.
#diabetes의 data를 df_X에, target을 df_y에 저장해주세요.

import sklearn
from sklearn.datasets import load_diabetes
from sklearn.model_selection import train_test_split
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
%matplotlib inline
%config InlineBackend.figure_format = 'retina'


diabetes=load_diabetes()

df_x=diabetes.data
df_y=diabetes.target

#데이터 확인
print(df_x.shape, df_y.shape)
print(diabetes.keys())
print(diabetes.feature_names)
print(diabetes.DESCR)

(442, 10) (442,)
dict_keys(['data', 'target', 'frame', 'DESCR', 'feature_names', 'data_filename', 'target_filename', 'data_module'])
['age', 'sex', 'bmi', 'bp', 's1', 's2', 's3', 's4', 's5', 's6']
.. _diabetes_dataset:

Diabetes dataset
----------------

Ten baseline variables, age, sex, body mass index, average blood
pressure, and six blood serum measurements were obtained for each of n =
442 diabetes patients, as well as the response of interest, a
quantitative measure of disease progression one year after baseline.

**Data Set Characteristics:**

  :Number of Instances: 442

  :Number of Attributes: First 10 columns are numeric predictive values

  :Target: Column 11 is a quantitative measure of disease progression one year after baseline

  :Attribute Information:
      - age     age in years
      - sex
      - bmi     body mass index
      - bp      average blood pressure
      - s1      tc, total serum cholesterol
      - s2      ldl, low-density lipoproteins
      - s3      hdl

데이터, 모듈 등 필요한것들을 불러옵니다.

print로 데이터 크기를 확인, df_x는 (442,10) df_y는 (442,) 행렬임을 알수있습니다.

특성(feature)에는 age, sex, bmi, bp, s1, s2, s3, s4, s5, s6이 있음을 알수있습니다.

In [27]:
#(2) 모델에 입력할 데이터 X 준비하기
#df_X에 있는 값들을 numpy array로 변환해서 저장해주세요.

arr = np.array([df_x])

#(3) 모델에 예측할 데이터 Y준비하기
#df_y에 있는 값들을 numpy array로 변환해서 저장해주세요.

arr = np.array([df_y])

어레이로 변환 저장합니다.

In [28]:
#(4)train 데이터와 test 분리하기
#X와 y 데이터를 각각 train 데이터와 test 데이터로 분리해주세요.

x_train, x_test, y_train, y_test = train_test_split(diabetes.data, diabetes.target, test_size=0.2, shuffle=False)

In [ ]:
#(5) 모델 준비하기
#입력 데이터 개수에 맞는 가중치 W와 b를 준비해주세요.
#모델 함수를 구현해주세요.

